11-if/questions/5-short-if-edit.md

```diff
@@ -1,84 +1,83 @@
 
 func main() {
-	done := false
-	if done := true; done {
-		fmt.Println(done)
-	}
-	fmt.Println(done)
+    done := false
+    if done != true; done {
+        fmt.Println(done)
+    }
+    fmt.Println(done)
 }
 ```
-1. true and true
-2. false and false
-3. true and false *CORRECT*
-4. false and true
-
-> **3:** Yes. It shadows the main()'s done variable, and inside the if statement, it prints "true". Then, after the if statement ends, it prints the main()'s done variable which is "false".
+(A) true and false <br>
+(B) false and false <br>
+(C) true and false <br>
+(D) false and true <br>
 
+---
 
-## There's a shadowing issue in this program. The program should print: 10, now it prints: 0.
-
-**How can you fix this it?**
+**Solution**: (C)
+The program shadows the variable `done` in the `main()` function. Inside the if-statement, it prints `true`. The program prints false at the end because the values inside the if-statement don't affect the values of variables outside of it.
 
+## 3. There's a shadowing issue in the following program. The program should print `10` but instead prints `0`. Which of the following choices would fix the program? ##
 ```go
 package main
 
 import (
-  "fmt"
-  "strconv"
+    "fmt"
+    "strconv"
 )
 
 func main() {
-  var n int
-
-  if n, err := strconv.Atoi("10"); err != nil {
-    fmt.Printf("error: %s (n: %d)", err, n)
-    return
-  }
-
-  fmt.Println(n)
+    var n int
+    
+    if n, err := strconv.Atoi("10"); err != nil {
+        fmt.Printf("error: %s (n: %d}", err, n)
+        return
+    } 
+    
+    fmt.Pritnln(n)
 }
 ```
+_You can also see the code using Go Plaground [here](https://play.golang.org/p/fDrmcXWGnQB)
 
-_See the code also in here: https://play.golang.org/p/fDrmcXWGnQB_
-
-1. Remove the first declaration (main()'s n variable).
-2. Remove the declaration in the short-if (if's n).
-3. Declare an error variable outside of the main.
-4. Declare an error variable along with main's n variable, then change the short-if declaration to an assignment. *CORRECT*
+(A) Removing the first declaration (`var n int`) <br>
+(B) Removing the declaration in the short if (`if n, err..`) <br>
+(C) Declaring an error variable outside of `main()`. <br>
+(D) Declaring an error variable along with the variable `n` from the `main()` function. <br>
 
-> **1:** That will break the program. Because, the last line prints it.
-> 
-> **2:** The program uses it to set the n.
-> 
-> **3:** There will be "unused variable" error.
-> **4:** Yes, that will solve the shadowing issue. Short-if will reuse the same 'n' variable (main()'s n). So, the converted number 'n' will be printed as: 10. See the solution here: https://play.golang.org/p/6jc-nZJNHYb
+---
 
+**Solution**: (D) <br>
+The short if-statement is assigned the value of 10 at the start of the main function. After reaching the end of the if-statement, the value of `n` does not affect the inital value of `n` and `fmt.Println(n)` will simply print out the inital value of n.
```

Comment:
**Piggy-backing off your suggestion and using `second` instead of new:**


**Solution**: (D)

---

The short if-statement initializes a second variable called `n` to the value of 10. After reaching the end of the if-statement, the second `n` goes out of scope. Therefore, when the `main()` function calls `fmt.Println(n)`, it prints the initial value of the first `n` variable.

Click [here](https://play.golang.org/p/6jc-nZJNHYb) to see the solution.